### ISSUE TITLE
Parse any character until

### ISSUE THREAD

**Issue by vincent-lg** [dashbitco/nimble_parsec#134]
I cannot find the way to parse any character (unicode included). I have two different cases where it seems I have no other option:

- An identifier: in my parsing, an identifier is not just a group of ASCII letters, it could include UTF8 letters too. That's an interesting use case for regular expressions, since Unicode letters are somewhat supported. Lacking this, a kind of `anychar` would be appreciated. I just cannot find a way to do it, though I've tried to play with `lookahead_not` since the identifier is broken by a space in any case.
- The content of a string (`""`). I imagine it's not true that string parsing can only contain ASCII symbols. Again, the use case would be to parse until the closing quote sign (triple-quote for triple-quoted string). And again, I don't find a good way to do that.

Chances are I'm missing something or that `NimbleParsec` was never meant to parse a scripting language. There are two functions for utf8, but one has to specify the range each time, which is not very helpful with unicode containing a lot of possible entries.

Thanks for the clarification. And sorry if this is not "issue worthy".

**Comment by josevalim** [comment]
Wouldn't it be `utf8_string(not: ?")`?

**Comment by vincent-lg** [comment]
Unfortunately, it doesn't seem to do quite what I want.
Identifiers should be utf-8 letters. I was at a loss about doing it without a regex or at least a true function I could call and which would make sure the string would not be parsed more than necessary. `aérien + 5` should accept everything before the space (which is not a valid letter).
I thought I had found the solution with `repeat_while`:

```elixir
  vexpr_var =
    repeat_while(
      utf8_char([]),
      {:check_varname, []}
    )
    |> unwrap_and_tag(:var)
    |> label("variable")

  defp check_varname(name, context, _, _) do
    case String.match?(name, ~r/^[[:alpha:]]+$/u) do
      true -> {:cont, context}
      false -> {:halt, context}
    end
  end
```
But the `check_varname` function seems to be called only once, after all the characters in the strings have been read. Not exactly what I'd like.
I also tried to play with `post_traverse`, but I guess this won't do since I need to validate the string.
Perhaps there's no solution. Probably something I just miss.

**Comment by josevalim** [comment]
You have to build the whole list. You can see how we deal with this in Elixir's syntax highlighter: https://github.com/elixir-makeup/makeup_elixir/blob/master/lib/makeup/lexers/elixir_lexer/atoms.ex.exs :)
